wait/awaitility.go

```diff
@@ -41,220 +34,151 @@ const (
+				return false, nil
+			}
+			return false, err
 		}
-	}
-	return newAwait
+		a.T.Logf("found service '%s'", metricsSvc.Name)
+		return true, nil
+	})
+	return *metricsSvc, err
 }
 
-// WaitForToolchainCluster waits until there is a ToolchainCluster representing a operator of the given type
+// WaitForToolchainClusterWithCondition waits until there is a ToolchainCluster representing a operator of the given type
 // and running in the given expected namespace. If the given condition is not nil, then it also checks
 // if the CR has the ClusterCondition
-func (a *SingleAwaitility) WaitForToolchainCluster(clusterType cluster.Type, condition *v1alpha1.ToolchainClusterCondition) error {
+func (a *Awaitility) WaitForToolchainClusterWithCondition(clusterType cluster.Type, namespace string, condition *v1alpha1.ToolchainClusterCondition) (v1alpha1.ToolchainCluster, error) {
 	timeout := a.Timeout
 	if condition != nil {
 		timeout = ToolchainClusterConditionTimeout
 	}
-	return wait.Poll(a.RetryInterval, timeout, func() (done bool, err error) {
-		_, ok, err := a.GetToolchainCluster(clusterType, condition)
-		if ok {
+	var c v1alpha1.ToolchainCluster
+	err := wait.Poll(a.RetryInterval, timeout, func() (done bool, err error) {
+		var ready bool
+		if c, ready, err = a.GetToolchainCluster(clusterType, namespace, condition); ready {
 			return true, nil
 		}
-		a.T.Logf("waiting for availability of %s ToolchainCluster CR (in namespace %s) representing operator running in namespace '%s'", clusterType, a.Ns, a.OtherOperatorNs)
+		a.T.Logf("waiting for availability of %s ToolchainCluster CR (in namespace %s) representing operator running in namespace '%s'", clusterType, a.Namespace, a.Namespace)
 		return false, err
 	})
+	return c, err
 }
 
-// WaitForToolchainClusterConditionWithName waits until there is a ToolchainCluster with the given name
+// WaitForNamedToolchainClusterWithCondition waits until there is a ToolchainCluster with the given name
 // and with the given ClusterCondition (if it the condition is nil, then it skips this check)
-func (a *SingleAwaitility) WaitForToolchainClusterConditionWithName(name string, condition *v1alpha1.ToolchainClusterCondition) error {
+func (a *Awaitility) WaitForNamedToolchainClusterWithCondition(name string, condition *v1alpha1.ToolchainClusterCondition) (v1alpha1.ToolchainCluster, error) {
 	timeout := a.Timeout
 	if condition != nil {
 		timeout = ToolchainClusterConditionTimeout
 	}
-	return wait.Poll(a.RetryInterval, timeout, func() (done bool, err error) {
-		cluster := &v1alpha1.ToolchainCluster{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Ns, Name: name}, cluster); err != nil {
+	c := v1alpha1.ToolchainCluster{}
+	err := wait.Poll(a.RetryInterval, timeout, func() (done bool, err error) {
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, &c); err != nil {
```

Comment:
We need to create a new empty instance of the ToolchainCluster for every pool loop
```suggestion
	err := wait.Poll(a.RetryInterval, timeout, func() (done bool, err error) {
	        c = v1alpha1.ToolchainCluster{}
		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, &c); err != nil {
```

Reply:
ah thanks, I missed that one!